--- ./arch/arm/boot/dts/am33xx.dtsi
+++ ./arch/arm/boot/dts/am33xx.dtsi
@@ -781,7 +781,17 @@
 				interrupt-names = "ecap0";
 				status = "disabled";
 			};
-
+            
+			eqep0: eqep@0x48300180 {
+				compatible = "ti,am33xx-eqep";
+				reg = <0x48300180 0x80>;
+				clocks = <&l4ls_gclk>;
+				clock-names = "fck";
+				interrupt-parent = <&intc>;
+				interrupts = <79>;
+				status = "disabled";
+			};
+            
 			ehrpwm0: pwm@48300200 {
 				compatible = "ti,am3352-ehrpwm",
 					     "ti,am33xx-ehrpwm";
@@ -816,6 +826,16 @@
 				status = "disabled";
 			};
 
+  			eqep1: eqep@0x48302180 {
+				compatible = "ti,am33xx-eqep";
+				reg = <0x48302180 0x80>;
+				clocks = <&l4ls_gclk>;
+				clock-names = "fck";
+				interrupt-parent = <&intc>;
+				interrupts = <88>;
+				status = "disabled";
+			};          
+            
 			ehrpwm1: pwm@48302200 {
 				compatible = "ti,am3352-ehrpwm",
 					     "ti,am33xx-ehrpwm";
@@ -849,6 +869,16 @@
 				interrupt-names = "ecap2";
 				status = "disabled";
 			};
+            
+			eqep2: eqep@0x48304180 {
+				compatible = "ti,am33xx-eqep";
+				reg = <0x48304180 0x80>;
+				clocks = <&l4ls_gclk>;
+				clock-names = "fck";
+				interrupt-parent = <&intc>;
+				interrupts = <89>;
+				status = "disabled";
+			};
 
 			ehrpwm2: pwm@48304200 {
 				compatible = "ti,am3352-ehrpwm",

